modules/docs/mdx/5.0-MIGRATION-GUIDE.mdx

```diff
@@ -621,3 +663,346 @@ we've made a few updates to the component in v5:
+several subcomponents like `Popup.Target` and `Popup.CloseButton`. These components are hooked up to
+the `PopupModel` via React context and have access to state and events. `Popup.Card` is what the v4
+`Popup` once was.
+
+All behavior hooks, like `useCloseOnEscape` now take a `model` instead of variable parameters. This
+allowed us to fix some subtle bugs. Using the `PopupModel` means all hooks have access to all Popup
+state and events without passing in many parameters.
+
+#### usePopup
+
+In the above example, `usePopup` has been removed. `usePopupModel` should be used instead. All
+subcomponents have an associated behavior hook. For example, `Popup.Target` uses a hook called
+`usePopupTarget`. If you need to use your own components for any reason, these hooks are available.
+`Popup.Target` and `Popup.CloseButton` do not include any styling. They both use `SecondaryButton`
+as the default component rendered. You can change this via the `as` prop. For example, the following
+will render an unstyled button:
+
+```tsx
+<Popup.Target as="button">Show</Popup.Target>
+```
+
+Pass a `css` prop, or a styled button instead to have a custom styled button. You could even pass
+`IconButton` if you need an icon button to show a popup instead!
+
+If you were using `usePopup`, here's a list of equivalent APIs:
+
+| Before                                                                  | After                                 |
+| ----------------------------------------------------------------------- | ------------------------------------- |
+| `const { popperProps, targetProps, closePopup, stackRef } = usePopup()` | `const model = usePopupModel()`       |
+| `popperProps.open`                                                      | `model.state.visibility !== 'hidden'` |
+| `closePopup()`                                                          | `model.events.hide()`                 |
+| `stackRef` or `popperProps.ref`                                         | `model.state.stackRef`                |
+| `popperProps.anchorElement`                                             | `model.state.targetRef.current`       |
+| `targetProps.onClick`                                                   | `usePopupTarget(model).onClick`       |
+
+#### New Focus management
+
+A common theme we noticed in uses of Popup in the wild was focus management. Developers were
+manually passing a `ref` to the target button element and manually returning focus to it when
+closing a Popup through whatever means it was closed. This use-case should be handled now by a new
+hook: `useReturnFocus`. By default, `useReturnFocus` will return focus to the `targetRef` in the
+model, which is set by `Popup.Target`. This can be overridden via `returnFocusRef` passed to the
+model on creation. This `returnFocusRef` might make some migrations easier if `Popup.Target` cannot
+be used for whatever reason.
```

Comment:
```suggestion
#### New Focus Management

A common theme we noticed in uses of Popup in the wild was focus management. Developers were
manually passing a `ref` to the target button element and manually returning focus to it when
closing the Popup. This use case should now be handled by the new `useReturnFocus` hook. By default,
`useReturnFocus` will return focus to the `targetRef` in the model, which is set by `Popup.Target`.
This can be overridden by passing `returnFocusRef` to the model on creation. `returnFocusRef` should
make your migration easier if `Popup.Target` cannot be used for whatever reason.
```